### ISSUE TITLE
How many epochs to get FID <10

### ISSUE THREAD

**Issue by AhmadObeid** [cvlab-stonybrook/PathLDM#18]
Dear authors, 
thanks again for sharing your code. 
I am wondering how many epochs did it take you to reach the reported FID?
I have been training for a while now. The lowest FID I got is 240, and then it went up and down and is now back to 309.
![W B Chart 5_21_2024, 1 21 45 PM](https://github.com/cvlab-stonybrook/PathLDM/assets/48792491/fba9f515-3302-41a5-bae0-684721d7b7d9)
I am training on two A100s with a batchsize of 200, and using the same config files provided in the repo.

Can you please share any thought on this?
Thank you,

**Comment by srikarym** [comment]
Can you check the generated samples to see if they are meaningful? FID ~300 tells me the model is not learning. This could be a bug in the dataloader.

**Comment by AhmadObeid** [comment]
Thanks for your answer. Below are the latest images (along with other things for your reference) that are obtained at around epoch 110. As you can see reconstruction seems to be excellent. 

**Reconstruction:**
![reconstruction_gs-570000_e-000137_b-001828](https://github.com/cvlab-stonybrook/PathLDM/assets/48792491/8f923030-9964-4e10-87c2-b9bc4d22efbd)

**Progressive:**
![progressive_row_gs-570000_e-000137_b-001828](https://github.com/cvlab-stonybrook/PathLDM/assets/48792491/37597572-e6da-424c-b8d3-4f223cff898d)

**Diffusion:**
![diffusion_row_gs-570000_e-000137_b-001828](https://github.com/cvlab-stonybrook/PathLDM/assets/48792491/1baf333e-825e-454b-8c5d-36edd131d293)

**Samples:**
![samples_gs-570000_e-000137_b-001828](https://github.com/cvlab-stonybrook/PathLDM/assets/48792491/2efe0537-e73f-4fb5-ba46-d9892110496b)

**Conditioning:**
![conditioning_gs-570000_e-000137_b-001828](https://github.com/cvlab-stonybrook/PathLDM/assets/48792491/b2b213de-7ea0-4046-8eda-42e5ea11617d)

**Inputs:**
![inputs_gs-570000_e-000137_b-001828](https://github.com/cvlab-stonybrook/PathLDM/assets/48792491/1c73fe29-132f-4cb6-8069-dc3efa8d7354)

**Comment by srikarym** [comment]
Reconstruction being excellent just means the VAE can decode the image back from the latent. Since VAE remains frozen, reconstruction is guaranteed to be decent.
Samples are not meaningful, though. What does the loss look like? Did you initialize the U-Net from Imagenet weights?

**Comment by bhosalems** [comment]
I have had similar problems. After using exact batchsize as author's I was able to get FID of 30 ish in 3rd epoch. If you do change the batch size effective batch size changes, we have to then tune learning rate accordingly.
